maker_dao/src/agent.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  Finding,
+  Alert,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+  AlertsResponse,
+  AlertQueryOptions,
+} from "forta-agent";
+
+import {
+  L1_ESCROW_ADDRESS_OPTIMISM,
+  L1_ESCROW_ADDRESS_ARBITRUM,
+  L2_DAI_GATEWAY_ARB,
+  L1_AARBITRUM_GATEWAY,
+  TRANSFER_EVENT,
+  DEPOSIT_FINALISED_EVENT,
+  ETHER_CHAINID,
+  OP_CHAINID,
+  ARBI_CHAINID,
+  GET_DAI_BALANCE,
+  GET_TOTAL_SUPPLY,
+} from "./constant";
+
+const PROVIDER = getEthersProvider();
+
+const alert: Alert = {
+  alertId: "L1_ESCROW",
+  chainId: 1,
+  hasAddress: () => true,
+  metadata: {
+    totalSupply: Number,
+    network: "Ethereum",
+  },
+};
+
+
+const emptyAlertResponse: AlertsResponse = {
+  alerts: [alert],
+  pageInfo: { hasNextPage: false },
+};
+
+const query: AlertQueryOptions = { alertIds: ["L1_ESCROW"] };
+
+export const provideHandleTransaction =
+  (
+    provider: ethers.providers.JsonRpcProvider,
+    contractAddr: string,
+    getL1Alert: (alertQuery: AlertQueryOptions) => Promise<AlertsResponse>
+  ) =>
+  async (txEvent: TransactionEvent) => {
+    let findings: Finding[] = [];
+    const transferEvent = txEvent.filterLog(TRANSFER_EVENT);
+
+    const network = await provider.getNetwork();
+
+    const chainId = network.chainId;
+    const DepositEvent = txEvent.filterLog(DEPOSIT_FINALISED_EVENT);
+
+    if (chainId !== 1) {
+      for (const deposit of DepositEvent) {
```

Comment:
```suggestion
    if (chainId !== 1) {
      const depositEvent = txEvent.filterLog(DEPOSIT_FINALISED_EVENT);
      for (const deposit of depositEvent) {
```
Since you'll only need one for each check, you can filter for the needed events _after_ checking the `chainId`.